airflow/api_fastapi/views/public/variables.py

```diff
@@ -56,3 +56,27 @@ async def get_variable(
         raise HTTPException(404, f"The Variable with key: `{variable_key}` was not found")
 
     return VariableResponse.model_validate(variable, from_attributes=True)
+
+
+@variables_router.patch("/{variable_key}", responses=create_openapi_http_exception_doc([400, 401, 403, 404]))
+async def patch_variable(
+    variable_key: str,
+    patch_body: VariableBody,
```

Comment:
Do I see it right that the variable key is sent with the URL and is additionally/redundantly contained in the body of the POST? (I see at least this is consistent to the existing API...) Would we need to cross check the content for consistency? Or is it a feature that I have "a" as key in the URL abd "b" in the body?

Reply:
Exactly. This is how things are usually done and are done also in the Airflow legacy API.

This is why we have the exact check you are describing in the view implementation here:
https://github.com/apache/airflow/pull/42929/files#diff-f33e731b8c0fedccbfd387d2fefa24f4148d99077292a4b37e7b9744372c7a9eR69